packages/types/src/async-utils.ts

```diff
@@ -0,0 +1,18 @@
+import { OnExecuteDoneEventPayload, OnExecuteDoneHookResult, OnExecuteDoneHookResultOnNextHook } from '@envelop/core';
+import isAsyncIterable from 'graphql/jsutils/isAsyncIterable.js';
+
+export function handleMaybeStream(
+  payload: OnExecuteDoneEventPayload,
+  fn: OnExecuteDoneHookResultOnNextHook
+): void | OnExecuteDoneHookResult {
+  if (isAsyncIterable(payload.result)) {
+    return { onNext: fn };
+  } else {
+    fn({
+      result: payload.result,
+      setResult: payload.setResult,
```

Comment:
Is `setResult` be optional?

In the case with our custom Logger plugin in RedwoodJS, we want to log portions of the `result` -- options redacted data and some timing traces -- in `onExecuteDone` but not necessarily modify the result.

Reply:
`setResult` is injected into the hook but fully optional. You don't need to call it, only of you want to override the execution result.